### ISSUE TITLE
In IE data is taking previous state

### ISSUE THREAD

**Issue by Prasad-YS** [react-csv/react-csv#141]
Not Sure why, even though data is updated in CSVLINK component data is taking by previous state, when I click 1st time it's gives me blank CSV file, later on it takes the previous state data.
In chrome it is working fine, but in IE it's giving this issue, Can you please help me with this issue

**Comment by mriccid** [comment]
What version of the package are you using?

**Comment by Prasad-YS** [comment]
"react-csv": "1.0.19",

**Comment by Prasad-YS** [comment]
I have updated to latest version, still same issue persisting :(

**Comment by mriccid** [comment]
How are you providing state to the `CSVLink` component? How is the state change handled, in terms of new props being passed to this component?

**Comment by Prasad-YS** [comment]
Yes, data is being pass via props

**Comment by Prasad-YS** [comment]
MY code looks like below 

{console.log("csvfileName ",this.state.csvfileName)}
          <Tooltip title="Export to CSV">
          <CSVLink data={this.state.exportedData} headers={exportedheaders} // onClick={this.csvHandler} 
          onClick={() => {
        this.csvHandler();
        console.log("You click the link ",this.state.csvfileName);
                          }}
        filename={this.state.csvfileName} className="btn btn-primary" target="">


And the colsole o/p is 

" You click the link  undefined
csvfileName  Ticket Trend-Hourly_14Feb2019_11-50.csv "

1st it's taking onlick function then the render function state is updating

**Comment by mriccid** [comment]
How is your state being controlled in the component that is rendering `CSVLink`?

**Comment by Prasad-YS** [comment]
I have different tabs, on clicking on every tabs the state will get update

**Comment by khumphrey** [comment]
I'm having similar issues with the filename at the least in terms of CSVLink taking previous state...
```js
<CSVLink
                data={this.props.data}
                headers={makeHeaders()}
                className='btn'
                asyncOnClick
                filename={this.state.filename}
                onClick={(event, done) => {
                  this.setState({ filename: this.getFileName(new Date()) }, () => done());
                }}
                target='_blank'
              >
```

The filename is empty and there is no data...

**Comment by khumphrey** [comment]
Pretty sure the issue is [here ](https://github.com/react-csv/react-csv/blob/master/src/components/Link.js#L46) because it is using the based in data even though it is an async and should probably be getting refreshed (props) data..

**Comment by mccabemj** [comment]
@khumphrey PR (#186) has been merged. This should be fixed in v1.1.2
